frontend/src/components/camp/picasso/Picasso.vue

```diff
@@ -0,0 +1,414 @@
+            <v-btn absolute
+                   top
+                   right x-small
+                   dark text
+                   class="ec-event--btn rounded-sm"
+                   @click.prevent="on.click"
+                   @mousedown.stop=""
+                   @mouseup.stop="">
+              <v-icon x-small>mdi-pencil</v-icon>
+            </v-btn>
+          </template>
+        </dialog-activity-edit>
+
+        <h4 class="v-event-title">
+          {{ getActivityName(event) }}
+        </h4>
+
+        <!-- extendBottom: used for resizing -->
+        <div
+          v-if="editable && timed"
+          class="v-event-drag-bottom"
+          @mousedown.stop="extendBottom(event)" />
+      </template>
+    </v-calendar>
+
+    <v-snackbar v-model="isSaving" light>
+      <template v-if="patchError">
+        <v-icon>mdi-alert</v-icon>
+        {{ patchError }}
+      </template>
+      <template v-else>
+        <v-icon class="mdi-spin">mdi-loading</v-icon>
+        {{ $tc('global.button.saving') }}
+      </template>
+    </v-snackbar>
+  </div>
+</template>
+<script>
+import { toRefs } from '@vue/composition-api'
+import useDragAndDrop from './useDragAndDrop.js'
+import { isCssColor } from 'vuetify/lib/util/colorUtils'
+
+import DialogActivityEdit from '@/components/dialog/DialogActivityEdit.vue'
+
+export default {
+  name: 'Picasso',
+  components: {
+    DialogActivityEdit
+  },
+  props: {
+    // period for which to schow picasso
+    period: {
+      type: Object,
+      required: true
+    },
+
+    // list of scheduleEntries
+    scheduleEntries: {
+      type: Array,
+      required: true
+    },
+
+    // false disables drag & drop and disabled edit dialogs
+    editable: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+
+    // v-calendar start: starting date (first day)
+    start: {
+      type: Number,
+      required: true
+    },
+
+    // v-calender end: end date (last day)
+    end: {
+      type: Number,
+      required: true
+    },
+
+    // v-calendar type
+    type: {
+      type: String,
+      required: false,
+      default: 'custom-daily'
+    },
+
+    // v-calendar intervalHeight
+    intervalHeight: {
+      type: Number,
+      required: false,
+      default: 0
+    }
+
+  },
+
+  // emitted events
+  emits: [
+    'changePlaceholder', // triggered continuously while a new entry is being dragged (parameters: startTime, endTime)
+
+    'newEntry', // triggered once when a new entry was created via drag & drop (parameters: startTime, endTime)
+
+    'openEntry' // triggered when an existing entry is clicked on (parameters: entry, newTab)
+  ],
+
+  // composition API setup
+  setup (props, { emit }) {
+    const { editable } = toRefs(props)
+
+    const {
+      entryMouseDown,
+      timeMouseDown,
+      timeMouseMove,
+      timeMouseUp,
+      nativeMouseUp,
+      extendBottom,
+      isSaving,
+      patchError
+    } = useDragAndDrop(editable, emit)
```

Comment:
Ist es wichtig dass du diese Einträge hier spreadest? So muss diese Komponente wieder Wissen haben über die interne Funktionalität des DragAndDrop Composables. Könntest du es nicht stattdessen so machen:
```js
const dragAndDrop = useDragAndDrop(editable, emit)
return { ...dragAndDrop }
```

Reply:
Hab ich so mal aus den Tutorials kopiert.

Spread-Operator würde technisch sicher gehen. Ich finds so wie es jetzt ist eher expliziter. Das sind ja nicht die Interna des Composables sondern genau dessen public-Interface, was ich in der aktuellen Komponente nutzen will und im Template-Code so auch wieder verwende.

Wenn ich die Komponente Picasso nicht kenne und ich sehe, dass auf dem v-calendar die action `timeMouseMove` aufgerufen wird, dann finde ich jetzt relativ schnell heraus, dass diese aus dem useDragAndDrop hervorkommt. Ansonsten müsst ich in das Composable hineinschauen, um zu verstehen, dass `timeMouseMove` von dort herkommt.